path_collector.go

```diff
@@ -115,6 +110,20 @@ func (c *FilepathCollector) walkDirectoryForYaml(dir string) ([]string, error) {
 	return paths, err
 }
 
+func (c *FilepathCollector) extensionMatches(name string) bool {
+	for _, ext := range c.Extensions {
+		// Users may specify "yaml", but we only want to match ".yaml", not "buyaml".
+		if !strings.HasPrefix(ext, ".") {
+			ext = "." + ext
+		}
```

Comment:
What about this

```suggestion
		ext = "." + strings.TrimPrefix(ext, ".")
```

Reply:
These would be basically equivalent in functionality, so it's down to a matter of opinion. I'll probably keep what's here out of personal preference; I like the idea of checking if something is needed and then doing it vs finding a way to always do something.

Reply:
Either works.

IMHO with the `if` the code reads like you'd say it out loud: "if ext has no '.' prefix, add it". This makes the code easily readable.

The benefit of our suggestion is that it uses a single line and avoids a branch.

Since readability is _very_ important to me, I'd always take the more readable option, even if it ends up being a bit more wordy.

Reply:
I'm fine with what you said